oxbow/src/bed.rs

```diff
@@ -0,0 +1,291 @@
+                    Field::new("b", DataType::UInt8, true),
+                ],
+                vec![
+                    Box::new(UInt8Builder::new()),
+                    Box::new(UInt8Builder::new()),
+                    Box::new(UInt8Builder::new()),
+                ],
+            ),
+            block_sizes: ListBuilder::new(UInt64Builder::new()),
+            block_starts: ListBuilder::new(UInt64Builder::new()),
+        })
+    }
+}
+
+impl BatchBuilder for BedBatchBuilder {
+    // Noodles implements bed Record structs for bed3-bed9 and bed12.
+    // The following only handles bed12.
+    type Record<'a> = &'a bed::Record<12>;
+
+    fn push(&mut self, record: Self::Record<'_>) {
+        // Mandatory fields (spec)
+        self.reference_sequence_name
+            .append_value(record.reference_sequence_name());
+        self.start_position
+            .append_value(record.start_position().get().try_into().unwrap());
+        self.end_position
+            .append_value(record.end_position().get().try_into().unwrap());
+
+        // Optional fields (spec)
+        if let Some(name) = record.name() {
+            self.name.append_value(name.to_string());
+        } else {
+            self.name.append_null();
+        }
+
+        if let Some(score) = record.score() {
+            self.score.append_value(score.get().try_into().unwrap());
+        } else {
+            // A score of 0 will be handled as a null as well
+            self.score.append_null();
+        }
+
+        if let Some(strand) = record.strand() {
+            self.strand.append_value(strand.to_string());
+        } else {
+            self.strand.append_null();
+        }
+
+        // thick_start and thick_end are required fields in Noodle's bed7/8 structs.
+        self.thick_start
+            .append_value(record.thick_start().get().try_into().unwrap());
+
+        self.thick_end
+            .append_value(record.thick_end().get().try_into().unwrap());
+
+        if let Some(color) = record.color() {
+            let colors = [color.red(), color.green(), color.blue()];
+            for (i, val) in colors.iter().enumerate() {
+                self.color
+                    .field_builder::<UInt8Builder>(i)
+                    .unwrap()
+                    .append_value(*val);
+            }
+
+            // Handle 1 or 2 missing colors
+            if colors.len() == 2 {
+                self.color
+                    .field_builder::<UInt8Builder>(2)
+                    .unwrap()
+                    .append_null();
+            } else if colors.len() == 1 {
+                self.color
+                    .field_builder::<UInt8Builder>(1)
+                    .unwrap()
+                    .append_null();
+                self.color
+                    .field_builder::<UInt8Builder>(2)
+                    .unwrap()
+                    .append_null();
+            }
+            self.color.append(true);
```

Comment:
Another arrow builder gotcha is the difference between `append_value` and `append`. Not all builders have both, but `append` is for the struct and `append_value` (in the for loop above) is for the subarray